cdap-proto/src/main/java/io/cdap/cdap/proto/security/StandardPermission.java

```diff
@@ -21,9 +21,8 @@
  */
 public enum StandardPermission implements Permission {
   /**
-   * Create permission should be used when entity is created with a generated name.
-   * In this case it's checked on entity parent. If name is supplied beforehand,
-   * {@link #UPDATE} should be checked instead.
+   * Create an entity. Can be used both with enforceOnParent when name is not known beforehand and with
+   * enforce when name is provided.
```

Comment:
Should we be enforcing on the entity itself when it doesn't exist yet?

Reply:
Sounds good, though ideally we would want to eventually centralize our authorization model in such a way that this name is abstracted away from the extension (since it doesn't exist yet).